server/http_test.go

```diff
@@ -842,3 +954,477 @@ func TestGetSubscriptionsForChannel(t *testing.T) {
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID:      "mockTemplateID2___________",
+							Filters: getMockSubscriptionFilter("jira:issue_updated"),
+						},
+					}), t),
+		},
+		"Editing subscription template, GetProject mocked error. Existing subscription has existing project.": {
+			subscriptionTemplate: fmt.Sprintf(`{"instance_id": "jiraurl1", "id": "mockTemplateID2___________", "name": "subscription name", "channel_id": "channelaaaaaaaaaabbbbbbbbb", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
+			expectedStatusCode:   http.StatusInternalServerError,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
+				withExistingChannelSubscriptionTemplates(
+					[]SubscriptionTemplate{
+						{
+							ID:      "mockTemplateID2___________",
+							Filters: getMockSubscriptionFilter("jira:issue_updated"),
+						},
+					}), t),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			count++
+			fmt.Print(count)
+			api := &plugintest.API{}
+			p := Plugin{}
+
+			api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+			api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+			api.On("GetChannelMember", mockAnythingOfTypeBatch("string", 2)...).Return(&model.ChannelMember{}, (*model.AppError)(nil))
+			api.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(&model.Post{}, nil)
+
+			if tc.apiCalls != nil {
+				tc.apiCalls(api)
+			}
+
+			p.updateConfig(func(conf *config) {
+				conf.Secret = someSecret
+			})
+			p.SetAPI(api)
+			p.userStore = mockUserStore{}
+			p.instanceStore = p.getMockInstanceStoreKV(1)
+
+			w := httptest.NewRecorder()
+			request := httptest.NewRequest(http.MethodPut, "/api/v2/subscriptionTemplates", ioutil.NopCloser(bytes.NewBufferString(tc.subscriptionTemplate)))
+			if !tc.skipAuthorize {
+				request.Header.Set(HeaderMattermostUserID, model.NewId())
+			}
+			p.ServeHTTP(&plugin.Context{}, w, request)
+			assert.Equal(t, tc.expectedStatusCode, w.Result().StatusCode)
+		})
+	}
+}
+
+func TestCreateSubscriptionTemplate(t *testing.T) {
+	for name, tc := range map[string]struct {
+		subscriptionTemplate string
+		expectedStatusCode   int
+		skipAuthorize        bool
+		apiCalls             func(*plugintest.API)
+	}{
+		"Invalid": {
+			subscriptionTemplate: "{}",
+			expectedStatusCode:   http.StatusInternalServerError,
+		},
+		"Not Authorized": {
+			subscriptionTemplate: "{}",
+			expectedStatusCode:   http.StatusUnauthorized,
+			skipAuthorize:        true,
+		},
+		"Initial Subscription Template": {
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "some name", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
```

Comment:
```suggestion
			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "some name", "channel_id": "mockChannelID", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
```
instead of writing random text it's better to write meaningful dummy data